hub/management/commands/import_wildlife_trust_reserves.py

```diff
@@ -0,0 +1,104 @@
+from datetime import date
+
+from django.conf import settings
+
+import pandas as pd
+from tqdm import tqdm
+
+from hub.models import Area, AreaData, AreaType, DataSet
+
+from .base_importers import BaseAreaImportCommand
+
+
+class Command(BaseAreaImportCommand):
+    help = "Import data about wildlife trust reserves in each constituency"
+    data_file = settings.BASE_DIR / "data" / "wildlife_trust_reserves.csv"
+    defaults = {
+        "label": "Wildlife Trusts Reserves",
+        "description": "Names of Wildlife Trusts reserves in each constituency.",
+        "data_type": "json",
+        "category": "place",
+        "release_date": str(date.today()),
+        "source_label": "Data from the Wildlife Trusts.",
+        "source": "https://www.wildlifetrusts.org/",
+        "source_type": "api",
+        "data_url": "https://www.wildlifetrusts.org/jsonapi/node/reserve",
+        "table": "areadata",
+        "default_value": {},
+        "is_filterable": False,
+        "is_shadable": False,
+        "comparators": DataSet.comparators_default(),
+    }
+
+    count_defaults = {
+        "label": "Number of Wildlife Trusts Reserves",
+        "description": "Number of Wildlife Trusts reserves in each constituency.",
+        "data_type": "integer",
+        "category": "place",
+        "release_date": str(date.today()),
+        "source_label": "Data from the Wildlife Trusts.",
+        "source": "https://www.wildlifetrusts.org/",
+        "source_type": "api",
+        "table": "areadata",
+        "data_url": "https://www.wildlifetrusts.org/jsonapi/node/reserve",
+        "default_value": 0,
+        "is_filterable": True,
+        "is_shadable": True,
+        "comparators": DataSet.numerical_comparators(),
+    }
+
+    data_sets = {
+        "wildlife_trusts_reserves": {
+            "defaults": defaults,
+        },
+        "wildlife_trusts_reserves_count": {
+            "defaults": count_defaults,
+        },
+    }
+
+    def delete_data(self):
+        for data_type in self.data_types.values():
+            AreaData.objects.filter(data_type=data_type).delete()
+
+    def handle(self, quiet=False, *args, **kwargs):
+        self._quiet = quiet
+        self.add_data_sets()
+        self.delete_data()
+        self.process_data()
+        self.update_max_min()
+
+    def process_data(self):
+        df = pd.read_csv(self.data_file)
+
+        if not self._quiet:
+            self.stdout.write("Importing wildlife trusts reserves data")
+
+        # Group by the area, and add the data from there
+        for gss, data in tqdm(df.groupby("gss")):
+            try:
+                area = Area.objects.filter(
+                    area_type=AreaType.objects.get(code="WMC")
+                ).get(gss=gss)
+            except Area.DoesNotExist:
+                continue
+
+            json = []
+            for index, row in data.iterrows():
+                json.append(row[["title", "trust_name"]].dropna().to_dict())
```

Comment:
might be nice to have the URL for the reserve here too so we can link to it.

Reply:
I totally missed that this data was in the API!